### DISCUSSION TITLE
Reaction Times are inaccurate if the users navigate away from the page

### DISCUSSION THREAD

**Discussion by nikbpetrov** [jspsych/jsPsych#3186] [General]
jsPsych records reaction times by recording a timestamp at the beginning of a trial and a timestamp at the moment of response, using the Performance API. This is turbo-accurate in almost all cases (I've run masking experiments that require sub-16ms precision and have produced similar results to MATLAB).

However, if a user navigates away from a page (or similar behaviour that causes the browser to introduce throttling or sleeping), the RT can be skewed. In my data, I have found a difference of 45 hours. (RT of 48 hours was expected but a RT of only 3 was recorded).

The issues is much, much larger than jsPsych - see [this issue](https://github.com/mdn/content/issues/4713) - but overall it is centered around the fact that browsers handle "tab sleep" differently in relation to the Performance API - technically performance.now() should keep ticking in all cases, but it actually doesn't for some OS/browser variations.

@jodeleeuw might be worth adding a note about this in the Timing Accuracy page.

**Comment by nikbpetrov** [comment]
To add: This is perhaps not the end of the world, given that when such long RTs are allowed (e.g. participant takes a break), one can use the `time_elapsed` variable instead - it seems to be robust to tab throttles as it records the timestamp given by the browser. This has obvious limitations, solved on the ms level by the Performance API (e.g. clock changes, timezone changes, manual user changes etc), but all these are rare scenarios.

Overall, the fact that `time_elapsed` relies on the Date API, while the RTs rely on the Performance API seems to be a really cool, perhaps unintended, feature.

**Comment by jodeleeuw** [comment]
I'm hoping to eventually merge the interaction data into the main data object (#495), so that the data would contain information about focus and blur events on the document for a trial. This would give another way of measuring this. In the case that someone does navigate away, I wonder if we should replace the RT with some other value?

**Reply by nikbpetrov** [reply]
RT will be rendered meaningless anyways if they navigate away for long enough so the Performance API loses connection so RT remains fine.

#495 would also be related to #2868
